src/trio/_tests/test_util.py

```diff
@@ -198,11 +198,16 @@ def __init__(self, a: int, b: float) -> None:
             assert a == 8
             assert b == 3.15
 
+        def __reduce__(self) -> tuple[Any, ...]:  # type: ignore[explicit-any]
+            return ("dummy", ())
+
     with pytest.raises(TypeError):
         SpecialClass(8, 3.15)
 
     # Private constructor should not raise, and passes args to __init__.
-    assert isinstance(SpecialClass._create(8, b=3.15), SpecialClass)
+    inst = SpecialClass._create(8, b=3.15)
+    assert isinstance(inst, SpecialClass)
+    assert inst.__reduce__()[0] == SpecialClass._create
```

Comment:
might as well check that the other args are handled correctly as well
```suggestion
    inst = SpecialClass._create(8, b=3.15)
    assert isinstance(inst, SpecialClass)
    reduced = inst.__reduce__()
    assert isinstance(reduced[0](*reduced[1]), SpecialClass)
```

Reply:
I guess at that point we can just use `pickle`.